hudi-common/src/main/java/org/apache/hudi/common/table/timeline/MetadataConversionUtils.java

```diff
@@ -244,18 +246,44 @@ public static Option<HoodieCommitMetadata> getHoodieCommitMetadata(HoodieTableMe
     return Option.of(TimelineUtils.getCommitMetadata(hoodieInstant, timeline));
   }
 
-  public static org.apache.hudi.avro.model.HoodieCommitMetadata convertCommitMetadata(
-          HoodieCommitMetadata hoodieCommitMetadata) {
-    ObjectMapper mapper = new ObjectMapper();
-    // Need this to ignore other public get() methods
-    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-    org.apache.hudi.avro.model.HoodieCommitMetadata avroMetaData =
-            mapper.convertValue(hoodieCommitMetadata, org.apache.hudi.avro.model.HoodieCommitMetadata.class);
+  /**
+   * Convert commit metadata from json to avro.
+   */
+  public static <T extends SpecificRecordBase> T convertCommitMetadata(HoodieCommitMetadata hoodieCommitMetadata) {
+    if (hoodieCommitMetadata instanceof HoodieReplaceCommitMetadata) {
+      return (T) convertReplaceCommitMetadata((HoodieReplaceCommitMetadata) hoodieCommitMetadata);
+    }
+    hoodieCommitMetadata.getPartitionToWriteStats().remove(null);
+    org.apache.hudi.avro.model.HoodieCommitMetadata avroMetaData = JsonUtils.getObjectMapper().convertValue(hoodieCommitMetadata, org.apache.hudi.avro.model.HoodieCommitMetadata.class);
     if (hoodieCommitMetadata.getCompacted()) {
       avroMetaData.setOperationType(WriteOperationType.COMPACT.name());
     }
-    // Do not archive Rolling Stats, cannot set to null since AVRO will throw null pointer
-    avroMetaData.getExtraMetadata().put(HoodieRollingStatMetadata.ROLLING_STAT_METADATA_KEY, "");
-    return avroMetaData;
+    return (T) avroMetaData;
+  }
+
+  /**
+   * Convert replacecommit metadata from json to avro.
+   */
+  public static org.apache.hudi.avro.model.HoodieReplaceCommitMetadata convertReplaceCommitMetadata(HoodieReplaceCommitMetadata replaceCommitMetadata) {
+    replaceCommitMetadata.getPartitionToWriteStats().remove(null);
+    replaceCommitMetadata.getPartitionToReplaceFileIds().remove(null);
+    return JsonUtils.getObjectMapper().convertValue(replaceCommitMetadata, org.apache.hudi.avro.model.HoodieReplaceCommitMetadata.class);
+  }
+
+  /**
+   * Convert commit metadata from avro to json.
+   */
+  public static <T extends SpecificRecordBase> byte[] convertCommitMetadataToJsonBytes(T avroMetaData, Class<T> clazz) {
+    Schema avroSchema = clazz == org.apache.hudi.avro.model.HoodieReplaceCommitMetadata.class ? org.apache.hudi.avro.model.HoodieReplaceCommitMetadata.getClassSchema() :
+        org.apache.hudi.avro.model.HoodieCommitMetadata.getClassSchema();
```

Comment:
I see the instance of check for replace commit metadata and commit metadata everywhere, can we avoid that?